### ISSUE TITLE
GCop171 false alarm

### ISSUE THREAD

**Issue by sungam3r** [Geeksltd/GCop#286]
```csharp
private static int Test(bool? value)
{
    if (value.HasValue && value.Value) // GCop171 There is no need for calling .Value. Replace with value.HasValue && value.
        return 0;
    return 1;
}
```

The proposed version is not compiled:
```csharp
private static int Test(bool? value)
{
    if (value.HasValue && value) // CS0019 Operator '&&' cannot be applied to operands of type 'bool' and 'bool?'
        return 0;
    return 1;
}
```

**Comment by sungam3r** [comment]
Ultimately it would be more correct to offer such an option
```csharp
private static int Test(bool? value)
{
    if (value == true)
        return 0;
    return 1;
}
```

**Comment by marjanjavid** [comment]
GCop 171 message still has some problem. it shows this message:
"There is no need for calling .Value. Replace with There is no need for calling '.Value' and '.HasValue'. use '== true"

**Comment by marjanjavid** [comment]
As I said this issue is fixed on development branch, it seems it is not updated on master
